pkg/arvo/app/chat.hoon

```diff
@@ -340,201 +222,197 @@
   ?+  i.wir
     (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
   ::
-  ::  %circles
       %circles
-    ?>  ?=(%circles -.rum)
-      =/  cis
+    (handle-rumor-circles rum)
+  ::
+      %circle
+    (handle-rumor-circle wir rum)
+  ::
+  ==
+::
+::  +handle-rumor-circles
+::
+++  handle-rumor-circles
+  |=  rum=rumor:hall
+  ^-  (quip move _this)
+  ?>  ?=(%circles -.rum)
+  =/  cis
+    ?:  add.rum
+      (~(put in circles.str) cir.rum)
+    (~(del in circles.str) cir.rum)
+  =.  str
+    %=  str
+      circles  cis
+      peers
         ?:  add.rum
-          (~(put in circles.str.sta) cir.rum)
-        (~(del in circles.str.sta) cir.rum)
-      =/  str
-        %=  str.sta
-          circles  cis
-          peers
-            ?:  add.rum
-              (~(put by peers.str.sta) [our.bol cir.rum] ~)
-            (~(del by peers.str.sta) [our.bol cir.rum])
-        ==
-      :-  (send-chat-update [[%circles cis] str])
-      this(str.sta str)
+          (~(put by peers.str) [our.bol cir.rum] ~)
+        (~(del by peers.str) [our.bol cir.rum])
+    ==
+  :_  this(str str)
```

Comment:
Because you do `=. str` above, you can just do `this` here, that already contains the updated `str`. Same for everywhere else you do `this(str str)`

Reply:
Ping.